app/models/user.rb

```diff
@@ -133,7 +133,7 @@ def self.default_role_and_status_for_email(email)
     if default_challenge_manager?(email)
       %w[challenge_manager pending]
     else
-      %w[solver active]
+      %w[evaluator pending]
     end
```

Comment:
This logic needs to be modified a bit. If a new user has an existing EvaluatorInvitation, their role and status should be `[evaluator, pending]`, otherwise it can use the existing `default_challenge_manager?(email)` condition and default to `[solver, active]` like it currently does.
Additionally, we need to add an [after_create callback](https://guides.rubyonrails.org/active_record_callbacks.html#create-callbacks) on User to deal with the Invitation. This will only be called the first time the User record is created and it can handle multiple invitations to different challenges.
```
class User
  after_create :accept_evaluator_invitation
  
  private
  def accept_evaluator_invitation
    EvaluatorInvitation.where(email: self.email).each do |invite|
      ChallengePhasesEvaluator.create(challenge: invite.challenge, phase: invite.phase, user: self)
      invite.destroy
    end
  end
end
```

Reply:
@stepchud Okay I understand, I'll modify the logic here and add some tests to cover this - thanks!